x-pack/plugins/security_solution/server/lib/detection_engine/routes/rules/preview_rules_route.ts

```diff
@@ -209,6 +261,14 @@ export const previewRulesRoute = async (
           )
           .map((item) => item.message);
 
+        // Refreshes alias to ensure index is able to be read before returning
+        await context.core.elasticsearch.client.asInternalUser.indices.refresh(
+          {
+            index: previewRuleDataClient.indexNameWithNamespace(spaceId),
+          },
+          { ignore: [404] }
```

Comment:
hmm why are we ignoring 404 here? doesn't that mean that the preview index hasn't been properly created &  functionality won't work?

Reply:
At this point, the `404` error would show up if there were no alerts written to the index and we wouldn't have created it because no alerts were found with the preview query. I believe in the event of the index not being properly created, there are other error handlers that would throw still

Reply:
so if the first preview ever doesn't have any results, you get a 404, which is then suppressed? that's sounds like an edge case. I think it's conceptually not very good to suppress errors like this in general, so maybe we should see the users see the error in that case.. we could ask product

Reply:
We can definitely ask, but as I understand it, this is expected behavior as it wouldn't be an error from the user's perspective. It's more of an internal error dealing with the rule registry's design behavior of creating an index only if alerts exist. Users would get no data back and the resulting histogram would be empty which would be accurate from their POV and they need not know if the index was created or not. We can for sure discuss it further though